File: frontend/src/types/teamspeak.ts
export interface TSClient {
  clid: number
  cid: number
  nickname: string
}

export interface TSChannel {
  cid: number
  pid: number
  name: string
  clients: TSClient[]
}

export interface TSStatus {
  clients: TSClient[]
  channels: TSChannel[]
  client_count: number
  server_host: string
  configured: boolean
}

File: backend/app/schemas/teamspeak.py
from pydantic import BaseModel


class TSClientOut(BaseModel):
    clid: int
    cid: int
    nickname: str


class TSChannelOut(BaseModel):
    cid: int
    pid: int
    name: str
    clients: list[TSClientOut] = []


class TSStatusOut(BaseModel):
    clients: list[TSClientOut]
    channels: list[TSChannelOut]
    client_count: int
    server_host: str
    configured: bool
